model/common/src/icon4py/model/common/decomposition/mpi_decomposition.py

```diff
@@ -245,12 +247,12 @@ def exchange(self, dim: gtx.Dimension, *fields: Sequence[gtx.Field]):
         log.debug(f"exchange for {len(fields)} fields of dimension ='{dim.value}' initiated.")
         return MultiNodeResult(handle, applied_patterns)
 
-    def exchange_and_wait(self, dim: gtx.Dimension, *fields: tuple):
+    def exchange_and_wait(self, dim: gtx.Dimension, *fields: tuple) -> None:
         res = self.exchange(dim, *fields)
         res.wait()
         log.debug(f"exchange for {len(fields)} fields of dimension ='{dim.value}' done.")
 
-    def __call__(self, *args, **kwargs) -> MultiNodeResult | None:
+    def __call__(self, *args: Any, **kwargs: dict[str, Any]) -> MultiNodeResult | None:
```

Comment:
```suggestion
    def __call__(self, *args: Any, dim: gtx.Dimension, wait: bool = True) -> MultiNodeResult | None:
```
I think that would be better as we are ignoring all other kwargs anyway.